xlive/Blam/Engine/cseries/cseries.h

```diff
@@ -78,8 +78,17 @@ extern bool g_catch_exceptions;
 // ADDR_SERVER: file offset in h2server.exe
 // TYPE: function
 // __VA_ARGS__: arguments for the function we want to invoke
-#define INVOKE_BY_TYPE(_addr_client, _addr_server, _type, ...) \
+// NOTE:
+// if server or client is equal to 0 then we pass the same address for both (ONLY ON RELEASE BUILDS)
+// REASON:
+// optimizes out a cmov or related instruction when building release whenever we call a function that doesn't have a dedi address specified
+#ifdef NDEBUG
+#define INVOKE_BY_TYPE(_addr_client, _addr_server, _type, ...)	\
+Memory::GetAddress<_type>(_addr_client == 0 ? _addr_server : _addr_client, _addr_server == 0 ? _addr_client : _addr_server)(__VA_ARGS__)
```

Comment:
have you even measured the perf gain of this change? I honestly perceive it as pointless.